### PR TITLE
Repo clean up

### PR DESCRIPTION
**Pull request by gmbr3**
* Add security policy and Dependabot updates
* Move some stuff around so it makes sense where to find things
* Add missed licenses
* Update README format to Markdown

### CODE REVIEW

**Comment by gmbr3** [comment]
Requires 'Dependabot version updates' be enabled on GitHub
Requires 'Private vulnerability reporting' be enabled on GitHub

**Comment by gmbr3** [comment]
I've got one more thing to add

File: README.ABI [hunk 1]
```diff
@@ -1,4 +1,4 @@
-When using gnu-efi 4.0 with an application which was wrote
+(1) When using gnu-efi 4.0 with an application which was wrote
```
Comment (pbatard) [inline README.ABI]: Might fix the typo too: "wrote" &rarr; "written"

Comment (gmbr3) [reply README.ABI]: Done

**Comment by julian-klode** [comment]
This has caused very complex legal questions and renders gnu-efi unsuitable for use cases it was used in before, such as shim, which is BSD licensed.

While shim has partially forked gnu-efi, this means we will need to stay on the fork post the GPL-2.0 relicensing or switch to a different library that remains BSD-licensed.  @vathpela 

This may also have inadvertently relicensed files of which @gmbr3 and @ncroxon are not the sole copyright holders, and as such violated their copyright.

**Comment by julian-klode** [comment]
Notably the following files were relicensed:

g/elf_x86_64_efi.lds
g/elf_mips64el_efi.lds
g/elf_loongarch64_efi.lds
g/elf_ia64_efi.lds
g/elf_ia32_fbsd_efi.lds
g/elf_ia32_efi_local.lds
g/elf_ia32_efi.lds
g/elf_arm_efi.lds
g/elf_aarch64_efi_local.lds
g/elf_aarch64_efi.lds
g/crt0-efi-ia32-local.S

They do not contain any licensing statements and the general license of the project was believed to be the 3-clause BSD license.

```

    Redistribution and use in source and binary forms, with or without
    modification, are permitted provided that the following conditions
    are met:

    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.
    * Redistributions in binary form must reproduce the above
      copyright notice, this list of conditions and the following
      disclaimer in the documentation and/or other materials
      provided with the distribution.
    * Neither the name of Hewlett-Packard Co. nor the names of its
      contributors may be used to endorse or promote products derived
      from this software without specific prior written permission.

    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
    CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
    MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS
    BE LIABLE FOR ANYDIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
    PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
    PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
    TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
    THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
    SUCH DAMAGE.
```

**Comment by julian-klode** [comment]
And I guess most of apps/ and certainly the code that Peter contributed.

Now given that everything was (believed to be) BSD licensed before the relicensing is sort of fine, you did not replace any BSD license grants that needed preserving with GPL ones, but it's still a very tough spot to be put into trying to sort out what happens now and where gnu-efi remains usable.

**Comment by gmbr3** [comment]
We may also be able to drop the LGPL of inc for elf.h aswell per:
```
3. Object Code Incorporating Material from Library Header Files.
The object code form of an Application may incorporate material from a header file that is part of the Library. You may convey such object code under terms of your choice, provided that, if the incorporated material is not limited to numerical parameters, data structure layouts and accessors, or small macros, inline functions and templates (ten or fewer lines in length), you do both of the following:

a) Give prominent notice with each copy of the object code that the Library is used in it and that the Library and its use are covered by this License.

b) Accompany the object code with a copy of the GNU GPL and this license document.
```

**Comment by ncroxon** [comment]
Can we get a clean up patch for this issue?

**Pull request merged**